compose.go

```diff
@@ -357,14 +357,27 @@ func (g *Generator) buildNixContainer(service types.ServiceConfig, networkMap ma
 
 	for _, v := range service.Volumes {
 		if volume, ok := volumeMap[v.Source]; ok {
-			// Replace the Compose volume name with the actual Docker volume
-			// name (i.e., potentially prefixed with project).
-			//
-			// This is what we'll use to refer to the volume in the generated
-			// container config.
-			volumeParts := strings.Split(v.String(), ":")
-			volumeParts[0] = volume.Name
-			c.Volumes[volume.Name] = strings.Join(volumeParts, ":")
+			if v.Volume != nil && v.Volume.Subpath != "" {
+				// Handle compose volume subpath syntax through mount options.
+				mount := fmt.Sprintf(
+					"type=volume,source=%s,target=%s,volume-subpath=%s",
```

Comment:
The `volume-subpath` option is Docker specific: https://github.com/docker/cli/blob/f74cd147bbecbe3a7b37d4a28c8f3da09513be30/opts/mount.go#L134

Podman just uses `subpath`: https://docs.podman.io/en/latest/markdown/podman-run.1.html#mount-type-type-type-specific-option

Reply:
So either adjust the name based on runtime, or only allow when runtime == Docker.

Reply:
Nice find! Man, Podman really does go to great lengths to remain Docker compatible.